jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/MetaInfConfiguration.java

```diff
@@ -720,25 +718,21 @@ public FileVisitResult visitFile(Path f, BasicFileAttributes attrs) throws IOExc
      */
     public Collection<URL> getTlds(URI uri) throws IOException
     {
-        HashSet<URL> tlds = new HashSet<URL>();
-
+        HashSet<URL> tlds = new HashSet<>();
         URI jarUri = uriJarPrefix(uri, "!/");
-        URL url = jarUri.toURL();
-        JarURLConnection jarConn = (JarURLConnection)url.openConnection();
-        jarConn.setUseCaches(Resource.getDefaultUseCaches());
-        JarFile jarFile = jarConn.getJarFile();
-        Enumeration<JarEntry> entries = jarFile.entries();
-        while (entries.hasMoreElements())
-        {
-            JarEntry e = entries.nextElement();
-            String name = e.getName();
-            if (name.startsWith("META-INF") && name.endsWith(".tld"))
+        try (Resource.Mount mount = Resource.mount(jarUri);
+             Stream<Path> stream = Files.walk(mount.root().getPath()))
+        {
+            Iterator<String> it = stream
+                .map(Path::toString)
+                .filter(filename -> filename.startsWith("META-INF") && filename.endsWith(".tld"))
+                .iterator();
+            String prefix = jarUri.toString();
+            while (it.hasNext())
             {
-                tlds.add(new URL(jarUri + name));
+                tlds.add(new URL(prefix + it.next()));
```

Comment:
This can be done entirely with Path and URI.
Don't `map(Path::toString())`.
Let the Stream/Iterator be Path.
Then just use `path.toURI().toURL()` here.
This will also keep the encoding proper the entire way through, which this current code doesn't do.

Btw, this change was done in the MultiReleaseJarFile PR too.

Reply:
Done.